sql-plugin/src/main/340+-and-330db/scala/org/apache/spark/sql/rapids/arithmetic.scala

```diff
@@ -55,17 +56,37 @@ abstract class CudfBinaryArithmetic extends CudfBinaryOperator with NullIntolera
 }
 
 trait GpuAddSub extends CudfBinaryArithmetic {
-
   override def outputTypeOverride: DType = GpuColumnVector.getNonNestedRapidsType(dataType)
   val failOnError: Boolean
-
   override def columnarEval(batch: ColumnarBatch): Any = {
-    if (dataType.isInstanceOf[DecimalType]) {
-      (left.dataType, right.dataType) match {
+    val outputType = dataType
+    val leftInputType = left.dataType
+    val rightInputType = right.dataType
+    
+    if (outputType.isInstanceOf[DecimalType]) {
+      (leftInputType, rightInputType) match {
         case (DecimalType.Fixed(p1, s1), DecimalType.Fixed(p2, s2)) =>
-          val resultType = resultDecimalType(p1, s1, p2, s2)
+          val resultType = dataType.asInstanceOf[DecimalType]
```

Comment:
nit: This can just be `outputType.asInstanceOf[DecimalType]`

Reply:
good catch, missed it